bindings/swc_cli/src/commands/compile.rs

```diff
@@ -266,28 +266,32 @@ struct InputContext {
 #[swc_trace]
 impl CompileOptions {
     fn build_transform_options(&self, file_path: &Option<&Path>) -> anyhow::Result<Options> {
-        let base_options = Options::default();
-        let base_config = Config::default();
-
         let config_file = self.config_file.as_ref().map(|config_file_path| {
             ConfigFile::Str(config_file_path.to_string_lossy().to_string())
         });
 
-        let mut ret = Options {
-            config: Config { ..base_config },
+        let mut options = Options {
             config_file,
-            ..base_options
+            ..Options::default()
         };
 
         if let Some(file_path) = *file_path {
-            ret.filename = file_path.to_str().unwrap_or_default().to_owned();
+            options.filename = file_path.to_str().unwrap_or_default().to_owned();
         }
 
         if let Some(env_name) = &self.env_name {
-            ret.env_name = env_name.to_string();
+            options.env_name = env_name.to_string();
+        }
+
+        if let Some(source_maps) = &self.source_maps {
+            options.source_maps = Some(match source_maps.as_str() {
+                "false" => SourceMapsConfig::Bool(false),
+                "true" => SourceMapsConfig::Bool(true),
+                value => SourceMapsConfig::Str(value.to_string()),
```

Comment:
I'm curious if we need to validate if value is acceptable and warn?

Reply:
As mentioned above, the validation handling is passed down

https://github.com/swc-project/swc/blob/ef1d966b5746d97bfb5b27a7330915b8d86806dc/crates/swc/src/config/mod.rs#L255-L258

Reply:
that's assert: doesn't it panic instead of gracefully error out?

Reply:
Yes, it does. But it would also panic now, if an invalid value is set in the configuration file.

Reply:
I still think it should be handled and emit proper warning instead of error, but probably not a blocker.